core/solver/lower_trs_kernels.hpp

```diff
@@ -40,26 +40,42 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/base/executor.hpp>
 #include <ginkgo/core/matrix/csr.hpp>
 #include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/solver/lower_trs.hpp>
 
 
 namespace gko {
 namespace kernels {
 namespace lower_trs {
 
 
+#define GKO_DECLARE_LOWER_TRS_CHECK_TRANSPOSABILITY_KERNEL()            \
+    void perform_transpose(std::shared_ptr<const DefaultExecutor> exec, \
+                           bool &transposability)
+
+
+#define GKO_DECLARE_LOWER_TRS_INIT_STRUCT_KERNEL()                \
+    void init_struct(std::shared_ptr<const DefaultExecutor> exec, \
+                     std::shared_ptr<gko::solver::SolveStruct> &solve_struct)
```

Comment:
I think I cannot use a plain ptr here because i need to use the shared_ptr constructor to create a new instance of the SolveStruct so that there is a reference count.

Reply:
That makes sense,